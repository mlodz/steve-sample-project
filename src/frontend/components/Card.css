.Card {
  border: 1px solid black;
  border-radius: 5px;
  width: 500px;
  margin: 5px;
  padding: 0px;
  position: relative;

  display: grid;
  grid-template-rows: 40px 2fr 30px;
  grid-template-columns: 1fr;

  h1 {
    grid-row: 1 / 2;
    grid-column: 1 / 2;
  }
  body {
    grid-row: 2 / 3;
    grid-column: 1 / 2;
  }
  footer {
    grid-row: 3 / 4;
    grid-column: 1 / 2;
  }

  h1 {
    font-size: 1.3em;
    font-weight: 500;
    margin: 0;
    padding: 0.5em 1em 0.5em 1em;

    color: var(--main-color);
    background-color: var(--main-bg-color);

    border-radius: 5px 5px 0 0;
    text-align: left;

    border-bottom: 1px solid black;
  }

  .body {
    padding: 1em;
  }
  .footer {
    padding: 0.5em 1em 0.5em 1em;
    color: #888;
    background-color: var(--second-bg-color);
    border-radius: 0 0 5px 5px;
    text-align: right;
  }

}

.nowrap {
  text-wrap: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}
