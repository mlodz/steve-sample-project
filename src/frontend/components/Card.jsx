import {Children, useState, useCallback} from 'react';

import './Card.css';

export default function Card({children}) {
  let [title, body, footer] =  Children.toArray(children);
  const defaultTitle = 'Missing Title';
  const defaultBody = 'Missing Body';
  const defaultFooter = 'Missing Footer';
  return (
    <div className="Card">
      <h1 className="title nowrap">{title || defaultTitle}</h1>
      <div className="body">
        {body || defaultBody}
      </div>
      <div className="footer nowrap">
        {footer || defaultFooter}
      </div>
    </div>
  );
}
