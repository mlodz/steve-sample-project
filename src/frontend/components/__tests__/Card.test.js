import React from 'react';
import renderer from 'react-test-renderer';
import Card from '../Card';

it('Card renders title, body, and footer', () => {
  const component = renderer.create(
      <Card>
        <span>Day Of The Week</span>
        <div>Today is Tuesday</div>
      </Card>,
  );
  let actual = component.toJSON();
  expect(actual).toMatchSnapshot();
});
