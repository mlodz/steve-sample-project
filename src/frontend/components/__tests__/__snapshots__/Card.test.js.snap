// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders title, body, and footer 1`] = `
<div
  className="Card"
>
  <h1
    className="title nowrap"
  >
    <span>
      Day Of The Week
    </span>
  </h1>
  <div
    className="body"
  >
    <div>
      Today is Tuesday
    </div>
  </div>
  <div
    className="footer nowrap"
  >
    Missing Footer
  </div>
</div>
`;
